lib/rules/no-array-prototype-extensions.js

```diff
@@ -145,6 +145,41 @@ function variableNameToWords(name) {
     .split(' ');
 }
 
+/**
+ * Returns the fixing object if it can be fixable otherwise returns an empty array.
+ *
+ * @param {Object} callExpressionNode The call expression AST node.
+ * @param {Object} fixer The ESLint fixer object which will be used to apply fixes.
+ * @returns {Object|[]}
+ */
+function applyFix(callExpressionNode, fixer, context) {
+  const propertyName = callExpressionNode.callee.property.name;
+
+  switch (propertyName) {
+    case 'any':
+      return fixer.replaceText(callExpressionNode.callee.property, 'some');
+    case 'filterBy': {
+      const callArgs = callExpressionNode.arguments;
+      const sourceCode = context.getSourceCode();
+
+      return [
+        fixer.replaceText(callExpressionNode.callee.property, 'filter'),
+        fixer.insertTextBefore(sourceCode.ast, "import { get } from '@ember/object';\n"),
+        fixer.replaceText(
+          callArgs[0],
+          `item => get(item, ${sourceCode.getText(callArgs[0])}) === ${sourceCode.getText(
```

Comment:
Right. Currently, ember-data returns non-native proxies. I think for now it is safe to use `get` rather than directly accessing value on the object.

Also, i believe we would come with another lint rule which enforces devs to use a normal way of accessing object values along with auto-fixer.

Since all the lint rules will re-run once an autofix is applied, it shouldn't cause any issue.
cc: @bmish

Reply:
That's fair. I'm okay either way then. The [ember/no-get](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md) should help remove unnecessary `get` usages, while attempting to ignore known-proxy usages. Note that the `no-get` doesn't currently cleanup the unused import statement.